src/odemis/gui/cont/acquisition.py

```diff
@@ -450,6 +450,455 @@ def open_acquisition_dialog(self):
+        self._gauge_future_conn = ProgressiveFutureConnector(
+            future=self._acq_future,
+            bar=self._panel.gauge_cryosecom_acq,
+            label=self._panel.txt_cryosecom_left_time,
+            full=False,
+        )
+        self._acq_future.add_done_callback(self._on_acquisition_done)
+
+    @call_in_wx_main
+    def _on_acquisition_done(self, future):
+        """
+        Called when the acquisition process is
+        done, failed or canceled
+        """
+        self._acq_future = None
+        self._gauge_future_conn = None
+        self._tab_data.main.is_acquiring.value = False
+
+        # try to get the acquisition data
+        try:
+            data, exp = future.result()
+            self._reset_acquisition_gui(state="FINISHED")
+            logging.info("The acquisition is done")
+        except CancelledError:
+            logging.info("The acquisition was canceled")
+            self._reset_acquisition_gui(state="CANCELED")
+            return
+        except Exception as exp:
+            logging.exception("The acquisition failed")
+            self._reset_acquisition_gui(
+                text="Acquisition failed (see log panel).", state="FAILED"
+            )
+            return
+        # save the data
+        self._scheduled_future = self._executor.submit(self._export_data, future)
+        self._scheduled_future.add_done_callback(self._on_export_data_done)
+
+    def _reset_acquisition_gui(self, text=None, state=None):
+        """
+        Resets some GUI widgets for the next acquisition
+        text (None or str): a (error) message to display instead of the
+          estimated acquisition time
+        state (str): the state of the acquisition future
+        """
+        # hide/enable some widgets
+        self._panel.gauge_cryosecom_acq.Hide()
+        self._panel.btn_cryosecom_acqui_cancel.Hide()
+        self._panel.btn_cryosecom_acquire.Enable()
+        self._panel.btn_cryosecom_change_file.Enable()
+        self._panel.btn_acquire_overview.Enable()
+        self._panel.z_stack_chkbox.Enable()
+        self._panel.streams_chk_list.Enable()
+        self._panel.txt_cryosecom_left_time.Hide()
+        self._tab.streambar_controller.resume()
+        self._panel.Layout()
+
+        if state == "FINISHED":
+            self._panel.txt_cryosecom_est_time.Show()
+            self._panel.txt_cryosecom_est_time.SetLabel("Saving file ...")
+            # update the counter of the filename
+            self._config.fn_count = update_counter(self._config.fn_count)
+            # reset the storing directory to the project directory
+            self._config.last_path = self._config.pj_last_path
+        elif state == "FAILED":
+            self._panel.txt_cryosecom_est_time.Show()
+            self._panel.txt_cryosecom_est_time.SetLabel(text)
+        elif state == "CANCELED":
+            self._panel.txt_cryosecom_est_time.Show()
+            self._update_acquisition_time()
```

Comment:
Add a `else: raise ValueError(...)`. This way, if you make a typo in the state, it's easier to notice.
Also, to make the state even less prone to typos, you could use constants like `ST_FINISHED = "FINISHED"`.